x-pack/plugins/fleet/server/types/rest_spec/app.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const GenerateServiceTokenRequestSchema = {
+  query: schema.object({
+    remote: schema.boolean({ defaultValue: false }),
```

Comment:
nit: I personally found the query parameter for POST request confusing and not consistent with our other endpoints, should we move this to a body parameter?

Reply:
done, also replaced the callout
<img width="656" alt="image" src="https://github.com/elastic/kibana/assets/90178898/ba6fdb10-bd6a-4225-ad2e-aa83ae50f520">